superset/assets/src/visualizations/Calendar/Calendar.js

```diff
@@ -80,9 +81,11 @@ function Calendar(element, props) {
       calContainer.text(`Metric: ${verboseMap[metric] || metric}`);
     }
     const timestamps = metricsData[metric];
-    const extents = d3.extent(Object.keys(timestamps), key => timestamps[key]);
+    const extents = d3Extent(Object.keys(timestamps), key => timestamps[key]);
     const step = (extents[1] - extents[0]) / (steps - 1);
-    const colorScale = colorScalerFactory(linearColorScheme, null, null, extents);
+    const colorScale = getSequentialSchemeRegistry()
+      .get(linearColorScheme)
+      .createLinearScale(extents);
 
     const legend = d3.range(steps)
```

Comment:
also noting that you could swap out `d3Range` here and also get rid of `d3 v3` in this file 🎉  (also not blocking you just seem to be swapping some so pointing out the low hanging 🍒🍌 :)

Reply:
👍